source/frontend/src/app/modules/expenses/components/expense-form/expense-form.component.ts

```diff
@@ -55,59 +57,79 @@ export class ExpenseFormComponent implements OnInit {
     private fb: FormBuilder,
     currency: CurrencyService,
     expensesMonthService: ExpensesMonthService,
-    private categoriesHttpClient: CategoryHttpClient,
     private expensesHttpClient: ExpensesHttpClientService,
-    private userConnectionsHttpClient: UserConnectionHttpClient) {
+    userService: UserService) {
+    this.currentUser = userService.user;
+    this.isItemShared = this.checkIfItemShared(this.item);
     this.currencies = currency.currencies;
-    this.categoriesHttpClient.getAllCategories()
+    userService.categories$
       .subscribe((categories) => this.categories = categories)
-    this.userConnectionsHttpClient.getUserConnections()
+    userService.connections$
       .subscribe((connections) => {
         this.friends = connections
           .filter((c) => c.status === UserConnectionStatus.accepted)
           .map((c) => c.person as AmbiguousUser);
 
+        if (this.item != null) {
+          // we need to add missing persons from list of permitted ones to let user select them in lookup
+          if (!this.isItemShared && this.item.permittedPersons.length > 0) {
+            this.item.permittedPersons.forEach((p) => {
+              if (!this.friends.some(f => f.id === p.id)) {
+                this.friends.push(p);
+              }
+            })
+          }
```

Comment:
We need to perform it in populateValues method since constructor is called before assigning value to `item` variable